services/auth-app/pkg/service/service.go

```diff
@@ -115,9 +119,13 @@ func (a *AuthAppService) HandleCreate(w http.ResponseWriter, r *http.Request) {
 			w.WriteHeader(http.StatusForbidden)
 			return
 		}
-		ctx, err = a.authenticateUser(cid, gwc)
+		ctx, err = a.authenticateUser(userID, userName, gwc)
 		if err != nil {
 			sublog.Error().Err(err).Msg("error authenticating user")
+			if strings.Contains(err.Error(), "requested user does not match") {
```

Comment:
use a dedicated error type if you want to react on it. String matching will always end up in bugs :)